### PR TITLE
Documentation improvements and drop debian/

### PR DESCRIPTION
**Pull request by williamdes**
Ref #444

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -57,6 +57,7 @@ if (EXISTS "${CMAKE_SOURCE_DIR}/.git")
             "--git-dir=${CMAKE_SOURCE_DIR}/.git"
             describe
             --abbrev=4
+            --tags
```
Comment (williamdes) [inline src/CMakeLists.txt:60]: The last tag was not annotated, this fixes it
```diff
- v41-167-gfa5b
+ v42-136-gfa5b
```

**Comment by williamdes** [comment]
If you can add the following uncommon options to the kcov.1 file, that would help me not having to write in this uncommon format ?

I checked help output and the file
Missing:
```
 --gcov                  use gcov parser instead of DWARF debugging info
 --clang                 use Clang Sanitizer-coverage parser
 --dump-summary          dump a summary of coverage on stdout

 --output-interval=ms    Interval to produce output in milliseconds (0 to
                         only output when kcov terminates, default 5000)

 --debug=X               set kcov debugging level (max 31, default 0)

 --configure=key=value,... Manually set configuration values. Possible values:
                           bash-use-basic-parser=1    Enable simple bash parser
                           command-name=STR           Name of executed command
                           css-file=FILE              Filename of bcov.css file
                           high-limit=NUM             Percentage for high coverage
                           low-limit=NUM              Percentage for low coverage
                           merged-name=STR            Name of [merged] tag in HTML
                           coveralls-service-name=STR Service name for coveralls

 --patchelf=cmd          patchelf binary to use (for system mode),
                         default: patchelf

 --bash-tracefd-cloexec  Force children to not be traced by configuring the trace
                         fd as non-cloneable with LD_PRELOAD. Buggy on some systems

```

File: src/CMakeLists.txt (hunk 1 above)
Comment (SimonKagstrom) [reply src/CMakeLists.txt:60]: Ah, OK!

I guess it's soon time for a new release again.

**Comment by SimonKagstrom** [comment]
Now merged the documentation changes as #446, thanks for noting it!

**Comment by williamdes** [comment]
> Now merged the documentation changes as #446, thanks for noting it!

Thank you !

Only #437 is left before trying to update this into Debian

**Pull request merged**